types/FBProfile.d.ts

```diff
@@ -0,0 +1,67 @@
+type ProfileMap = {
+  firstName: string,
+  lastName: string,
+  middleName: string,
+  imageURL: string,
+  linkURL: string,
+  userID: string,
+  email: string,
+  name: string,
+};
+ 
+/**
+* Rappresent the current profile logged
+*/
+export class FBProfile {
+  
+  /**
+   * The user id
+   */
+  userID?: string;
+
+  /**
+   * The user's email.
+   * IMPORTANT: This field will only be populated if your user has granted your application the 'email' permission.
```

Comment:
According to the comments in the (Android) impl file, you actually cannot retrieve it via the profile, or?

Reply:
No, the email property is not inside the Profile implementation of the Android ([https://developers.facebook.com/docs/reference/androidsdk/current/facebook/com/facebook/profile.html/?locale=it_IT](url)). The only way to retrieve the email of the user is over GraphRequest. Maybe when Facebook will add the limited login in Android, they also add the email field in the Profile? Honestly, I don't know why this field is missing in Android.

Reply:
I think `-IOS` suffix could be also applied to this one to inform about limitation in Android.

Reply:
Sorry @bernard-kms, can you tell me where to place this suffix please?

Reply:
I mean the `email` field, since this field is only valid for iOS as you mentioned.

Reply:
Mmmm... Honestly, i don't think this is a correct tip because, if you are don't asking the correct permissions (i.e. the "email" permission) for the method **loginInWithPermissions** you can't retrieve the value in the field even in iOS. I think that the values of the Profile are dependent to the type of the auth request (and his options) and not to the platform.

Reply:
In that case we should have the value also on Android right? In such case I would stick with the suffix till we can maintain the same behavior in both platforms.